### ISSUE TITLE
Avatar and AvatarGroup a11y

### ISSUE THREAD

**Issue by Heydon** [iTwin/stratakit#1229]
## 1. Avatar should enforce alternative text

MUI leaves the user to implement an `aria-label`. There is no required prop for alternative text. Can we add one?

## 2. AvatarGroup must use list semantics

A group of avatars should be semantically grouped and enumerated (WCAG 1.3.1). This requires list (`<ul>`) semantics.

## 3. Render AvatarGroup according to source order

The visual and source orders of avatars in an AvatarGroup are currently contradictory, violating WCAG 1.3.2. Reverse the order so that the first avatar is the first in both source and visual presentation.

**Comment by mayank99** [comment]
Considering this resolved via #1417 and #1437, with a note that alt text is handled on consumer end.
